src/s3path.jl

```diff
@@ -46,10 +46,12 @@ end
 
 """
     S3Path()
-    S3Path(str; version::$(AbstractS3Version)=nothing, config::$(AbstractS3PathConfig)=nothing)
+    S3Path(str::AbstractString; version::$(AbstractS3Version)=nothing, config::$(AbstractS3PathConfig)=nothing)
+    S3Path(path::S3Path; isdirectory=missing, version=missing, config=missing)
```

Comment:
Why `missing` in this case but `nothing` in the others? Note that a simpler way to go about this is to reference the fields of the positional argument as the default values for the keyword arguments, e.g.:
```julia
function S3Path(path::S3Path; isdirectory=path.isdirectory, version=path.version,
                config=path.config)
    return S3Path(path.bucket, path.key; isdirectory, version, config)
end
```

Reply:
oh i love this

Reply:
i'm annoyed that i didn't come up with this myself, lol